test/test-test.js

```diff
@@ -107,6 +108,59 @@ buster.testCase("sinon-test", {
         }, "Error");
     },
 
+    "restores stub after promise resolves": function (done) {
+        var object = {};
+
+        var promise = instance(function () {
+            return new Promise(function (resolve) {
+                nextTick(function () { resolve(object); });
+            });
+        }).call({});
+
+        promise.then(function (result) {
```

Comment:
It also works with Mocha, Ava. I imagine that it probably works in most modern test runners